### DISCUSSION TITLE
How to use the language server for developing my extension

### DISCUSSION THREAD

**Discussion by HO-COOH** [microsoft/vscode-cpptools#6921] [Q&A] [answered]
I want to develop an extension for C++ editing enhancement. I want to utilize the existing C++ language server in this cpptool so I don't have to reinvent the wheel to parse the source files. But I still have no idea how to achieve that after reading all the documentation about the language server development tutorial on vscode's official site. Thanks for any help and information about it.

**Comment by michelleangela** [comment, marked as answer]
@HO-COOH

There is no API to use the IntelliSense features of the C++ extension.

There is the API v[scode-cpptools-api](https://github.com/Microsoft/vscode-cpptools-api) that allows build system extensions to provide IntelliSense configurations for consumers of Microsoft's C/C++ extension for VS Code.
